### ISSUE TITLE
VERY SERIOUS ISSUES WITH THIS ADDON TO DO BASIC TASKS ON TILES.

### ISSUE THREAD

**Issue by MichaelIsGreat** [darktrojan/newtabtools#175]
1) With the latest versions of this addon and/or Firefox, it has become impossible to add new titles the simplest way possible!!
Even the Option, Add tile option for this addon simply does not work properly. At least with the many tiles that I currently have (61 tiles in a table of 10 rows and 8 columns).
I used to drag and drop new tiles from the bookmarks sidebar (a button that I added with Customize on my user interface) that I opened at the left but this method does not seem to work anymore even though it was very easy to do before!!

2) I would like to be able to move at a very particular position among my set of tiles that are on the new tab any tile using my method 1) but also through a second method through your interface when you click on your wheel of options (The Option, Add a tile does not work!)

3) From a list of bookmarks in a particular folder of the bookmarks, I should be able to open this particular folder and add automatically all the bookmarks in the order listed in this folder onto the new tab using your addon.
That is an added feature that I would like to see added to your addon.

4) I have 10 rows and 8 columns and therefore I would like to be able to move very quickly the pinned tiles from one location to another by just dragging them on the new tab.
I also want to add new tiles from my left bookmarks sidebar very quickly by selecting them at the left and dragging them on the new tab page at the location of my choice.
Currently, all this is basically not possible to do with your addon.
More, I have yet to understand how to add and remove tiles using your addon even though it seems there is the Add tile option in Option but it simply does not work at all.

5) When I remove a particular tile at a particular location, I want the tiles next to it to fill the gap left by the removed tile, not another unpin tile come at the place of the tile that I have just removed.

6) The basic tasks described on "Customize the New Tab page | Firefox Help" at https://support.mozilla.org/en-US/kb/customize-new-tab-page simply do not work anymore, at least with the big number of tiles that I have (61 tiles in a table of 10 rows and 8 columns).
For instance, the basic task "Add a Tile from your bookmarks" simply does not work with your addon and that is a HUGE problem for me. Because I need to be able to add and move and remove tiles in seconds and I cannot do that anymore with your addon for reasons I do not know.

7) I do not know if it is the big number of tiles that I have (61 tiles in a table of 10 rows and 8 columns) that causes the following very serious issues but I should assume that it should not be the case considering the huge amount of RAM that I have.
For instance, when I press the pin on a tile to remove it, it completely freezes my web page and take ages to remove the tile.
Same crazy behavior when I want to just move a tile to a new location, it is basically impossible to do!!

8) (ADDED POINT TO SOLVE) When there is a URL for a particular tile that I would like to change and this tile is pinned on the New Tab, I should be able to easily change the URL. The strange fact is that there is NO easy option to do that.
When I go to Options, I can go to the tile and change its title because there is a Tile field but, to change its URL, it is not possible as there is no URL field!! The URL is shown but there is no means to change it to something else.
Just add a URL field in this Option where all the options are shown to change the URL of a particular tile that is pinned on the New Tab and then it will be possible to make such a simple change of the URL very easy to do.

Thanks please to plan to solve these various serious issues, they are so SERIOUS that your addon is now unusable as I cannot add, move or remove addons quickly and normally.

When these various issues will be fixed, it will be really an excellent addon.

By the way, I have Windows 7 Ultimate and Firefox 44. I also have an Intel core i7 quadcore and 8 GB of RAM.

Many thanks to the developer for this excellent addon. Highly recommended.

**Comment by darktrojan** [comment]
Everything's working fine here (that is, the code isn't broken), so I reckon you have some other problem going on. Check for other addons or themes that could be interfering with the new tab page.

**Comment by MichaelIsGreat** [comment]
Hello Geoff,

Thanks for your feedback. As far as I know, I do not think that I have any addon that would conflict with your addon. I join three jpgs that show the full list of addons that I currently have and you could check yourself to see if there are addons that are known to conflict with your addon but I do not think so.


Kind RegardsMichael
 

    On Friday, March 11, 2016 4:03 AM, Geoff Lankow <notifications@github.com> wrote:
 

 Everything's working fine here (that is, the code isn't broken), so I reckon you have some other problem going on. Check for other addons or themes that could be interfering with the new tab page.—

**Comment by MichaelIsGreat** [comment]
Hello Geoff,

Just an idea. You do not see any problems with your addon maybe because you did not test it with the big number of pinned tiles that I have, as I have 10 rows and 8 columns with 62 tiles pinned using your addon. This big number of pinned tiles might be the reason why I have had the problems I mentioned. And you do not have these problems maybe because you did not test how your addon behave with this big number of pinned tiles?
Thanks for your great addon.

Regards
Michael
    On Friday, March 11, 2016 4:03 AM, Geoff Lankow <notifications@github.com> wrote:
 

 Everything's working fine here (that is, the code isn't broken), so I reckon you have some other problem going on. Check for other addons or themes that could be interfering with the new tab page.—

**Comment by darktrojan** [comment]
Okay, I think I've found out what the problem is. There seems to be a bug in Firefox (at least, I can't see any reason for what it's doing) that gets exponentially worse as the number of tiles goes up. I'm working on a workaround.

**Comment by MichaelIsGreat** [comment]
Hello Geoff,

I am amazed by your awesome behavior! I asked you to check it out and you did!! Maybe 95% of other developers might not have been ready to check it out at all and might not have been as conscientious and dedicated to check it out as you are!! Kudos. Many thanks, Geoff!!
I knew there was a serious problem!! If it is due to a Firefox bug, could you make sure to inform them about it so that they will solve it later on.
Good job, Geoff, for checking it out. And thank you again.


King Regards
Michael
 

    On Sunday, March 20, 2016 1:37 AM, Geoff Lankow <notifications@github.com> wrote:
 

 Okay, I think I've found out what the problem is. There seems to be a bug in Firefox (at least, I can't see any reason for what it's doing) that gets exponentially worse as the number of tiles goes up. I'm working on a workaround.—

**Comment by pnlawrence** [comment]
Geoff, I've experienced the same problem with a smaller grid. Some of the points made by Michael about adding and editing tiles have been there ever since I started using this addon. I guess you don't have a beta testing group that thinks as we do. Difficult is not the word to use. Frustrating is more accurate. I'm to the point where, even though I've contributed, I'm ready to go back to the FF default.
